rewrite-maven/src/main/java/org/openrewrite/maven/AddManagedDependencyVisitor.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright 2021 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.maven;
+
+import lombok.RequiredArgsConstructor;
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Validated;
+import org.openrewrite.internal.lang.Nullable;
+import org.openrewrite.maven.internal.InsertDependencyComparator;
+import org.openrewrite.maven.tree.MavenMetadata;
+import org.openrewrite.semver.LatestRelease;
+import org.openrewrite.semver.Semver;
+import org.openrewrite.semver.VersionComparator;
+import org.openrewrite.xml.AddToTagVisitor;
+import org.openrewrite.xml.XPathMatcher;
+import org.openrewrite.xml.tree.Content;
+import org.openrewrite.xml.tree.Xml;
+
+import java.util.List;
+import java.util.Objects;
+
+import static java.util.Collections.emptyList;
+
+@RequiredArgsConstructor
+public class AddManagedDependencyVisitor extends MavenIsoVisitor<ExecutionContext> {
+    private static final XPathMatcher MANAGED_DEPENDENCIES_MATCHER = new XPathMatcher("/project/dependencyManagement/dependencies");
+
+    private final String groupId;
+    private final String artifactId;
+    private final String version;
+
+    @Nullable
+    private final String versionPattern;
+
+    @Nullable
+    private final String scope;
+
+    @Nullable
+    private final Boolean releasesOnly;
+
+    @Nullable
+    private final String type;
+
+    @Nullable
+    private final String classifier;
+
+    @Nullable
+    private VersionComparator versionComparator;
+
+    @Nullable
+    private String resolvedVersion;
+
+    @Override
+    public Xml.Document visitDocument(Xml.Document document, ExecutionContext ctx) {
+
+        Validated versionValidation = Semver.validate(version, versionPattern);
+        if (versionValidation.isValid()) {
+            versionComparator = versionValidation.getValue();
+        }
+
+        String versionToUse = findVersionToUse(ctx);
+        if (versionToUse.equals(existingManagedDependencyVersion())) {
```

Comment:
What happens in the new version is older than the current version?

Reply:
The intent here is:
- to not add a managed dependency tag if one already exists for the same version.  
- If a managed dependency tag exists in the pom then no action will be taken.  

there is nothing to prevent the recipe from managing a managed dependency to a different version